~~~
/home/pi/project/smart-doorlock/thread/main.c:12:6: error: conflicting types for ‘close’
   12 | void close(); // close door
      |      ^~~~~
In file included from /home/pi/project/smart-doorlock/thread/main.c:7:
/usr/include/unistd.h:353:12: note: previous declaration of ‘close’ was here
  353 | extern int close (int __fd);
      |            ^~~~~
/home/pi/project/smart-doorlock/thread/main.c: In function ‘cds’:
/home/pi/project/smart-doorlock/thread/main.c:82:10: warning: returning ‘int’ from a function with return type ‘void *’ makes pointer from integer without a cast [-Wint-conversion]
   82 |   return 1;
      |          ^
/home/pi/project/smart-doorlock/thread/main.c:86:10: warning: returning ‘int’ from a function with return type ‘void *’ makes pointer from integer without a cast [-Wint-conversion]
   86 |   return 1;
      |          ^
/home/pi/project/smart-doorlock/thread/main.c:96:3: warning: implicit declaration of function ‘set_brightness’; did you mean ‘get_brightness’? [-Wimplicit-function-declaration]
   96 |   set_brightness(curVal);
      |   ^~~~~~~~~~~~~~
      |   get_brightness
/home/pi/project/smart-doorlock/thread/main.c: In function ‘ultraSonic’:
/home/pi/project/smart-doorlock/thread/main.c:117:10: warning: returning ‘int’ from a function with return type ‘void *’ makes pointer from integer without a cast [-Wint-conversion]
  117 |   return 1;
      |          ^
/home/pi/project/smart-doorlock/thread/main.c:140:3: warning: implicit declaration of function ‘set_distance’; did you mean ‘get_distance’? [-Wimplicit-function-declaration]
  140 |   set_distance(travelTime / 58);
      |   ^~~~~~~~~~~~
      |   get_distance
/home/pi/project/smart-doorlock/thread/main.c: In function ‘bluetooth’:
/home/pi/project/smart-doorlock/thread/main.c:170:35: warning: returning ‘int’ from a function with return type ‘void *’ makes pointer from integer without a cast [-Wint-conversion]
  170 |  if (wiringPiSetup () < 0) return 1 ;
      |                                   ^
/home/pi/project/smart-doorlock/thread/main.c:173:3: warning: ‘return’ with no value, in function returning non-void
  173 |   return;
      |   ^~~~~~
/home/pi/project/smart-doorlock/thread/main.c:167:7: note: declared here
  167 | void* bluetooth(void* argv){
      |       ^~~~~~~~~
/home/pi/project/smart-doorlock/thread/main.c: At top level:
/home/pi/project/smart-doorlock/thread/main.c:197:6: error: conflicting types for ‘close’
  197 | void close(){
      |      ^~~~~
In file included from /home/pi/project/smart-doorlock/thread/main.c:7:
/usr/include/unistd.h:353:12: note: previous declaration of ‘close’ was here
  353 | extern int close (int __fd);
      |            ^~~~~
/home/pi/project/smart-doorlock/thread/main.c: In function ‘door’:
/home/pi/project/smart-doorlock/thread/main.c:231:3: warning: implicit declaration of function ‘waitpid’ [-Wimplicit-function-declaration]
  231 |   waitpid(pid, &status, 0);
      |   ^~~~~~~
/home/pi/project/smart-doorlock/thread/main.c: In function ‘init’:
/home/pi/project/smart-doorlock/thread/main.c:280:9: error: ‘false’ undeclared (first use in this function)
  280 |  done = false;
      |         ^~~~~
/home/pi/project/smart-doorlock/thread/main.c:280:9: note: each undeclared identifier is reported only once for each function it appears in
/home/pi/project/smart-doorlock/thread/main.c: At top level:
/home/pi/project/smart-doorlock/thread/main.c:301:6: warning: conflicting types for ‘set_brightness’
  301 | void set_brightness(int value){
      |      ^~~~~~~~~~~~~~
/home/pi/project/smart-doorlock/thread/main.c:96:3: note: previous implicit declaration of ‘set_brightness’ was here
   96 |   set_brightness(curVal);
      |   ^~~~~~~~~~~~~~
/home/pi/project/smart-doorlock/thread/main.c:315:6: warning: conflicting types for ‘set_distance’
  315 | void set_distance(int value){
      |      ^~~~~~~~~~~~
/home/pi/project/smart-doorlock/thread/main.c:140:3: note: previous implicit declaration of ‘set_distance’ was here
  140 |   set_distance(travelTime / 58);
      |   ^~~~~~~~~~~~
make[2]: *** [CMakeFiles/smart_doorlock.dir/build.make:82: CMakeFiles/smart_doorlock.dir/thread/main.c.o] 오류 1
make[1]: *** [CMakeFiles/Makefile2:211: CMakeFiles/smart_doorlock.dir/all] 오류 2
make: *** [Makefile:103: all] 오류 2

~~~

--- a/thread/main.c
+++ b/thread/main.c
@@ -8,8 +8,8 @@
 #include <time.h>
 #include <wiringSerial.h>
 
-void open();  // open door
-void close(); // close door
+void open_door();  // open door
+void close_door(); // close door
 void set_done(int value);
 int get_done();
 
@@ -27,12 +27,15 @@ pthread_mutex_t lock_done;
 pthread_mutex_t lock_door;
 
 pthread_t child_thread[3];
+
 int brightness;
 pthread_mutex_t lock_brightness;
 int get_brightness();
+void set_brightness(int value);
 int distance;
 pthread_mutex_t lock_dist;
 int get_distance();
+void set_distance(int value);
 
 int main(){
 	init();
@@ -53,8 +56,8 @@ int main(){
 		pthread_exit(NULL);
 	}
 	while(1){
-		open();
-		close();
+		open_door();
+		close_door();
 
 		set_done(1);
 		if(get_done() == 1) break;
@@ -79,11 +82,11 @@ void* cds(void* argv){
 	int adcChannel = 0;
 	if(wiringPiSetupGpio()  < 0 ){
 		printf("wiringPiSetup() is failed\n");
-		return 1; 
+		return NULL;
 	}
 	if ((i2c_fd = wiringPiI2CSetupInterface (I2C_DEV, SLAVE_ADDR_01)) < 0 ){
 	    printf("wiringPi2CSetup Failed: \n");
-		return 1;
+		return NULL;
 	}
 	printf("I2C start....\n"); 
 
@@ -100,8 +103,6 @@ void* cds(void* argv){
 		cnt++;
 		if(get_done() == 1) break;
 	}
-
-	return 0; 
 }
 
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
@@ -114,7 +115,7 @@ void* ultraSonic(void* argv){
 	if(wiringPiSetupGpio () == -1)
 	{
 		printf("Unable GPIO Setup"); 
-		return 1;
+		return NULL;
 	}
 	pinMode (TP, OUTPUT);
 	pinMode (EP, INPUT);
@@ -167,10 +168,10 @@ void serialWrite(const int fd, const unsigned char c)
 void* bluetooth(void* argv){
 	int fd_serial ; //UART2 파일 서술자
 	unsigned char dat; //데이터 임시 저장 변수
-	if (wiringPiSetup () < 0) return 1 ;
+	if (wiringPiSetup () < 0) return NULL;
 	if ((fd_serial = serialOpen (UART2_DEV, BAUD_RATE)) < 0){ //UART2 포트 오픈
 		printf ("Unable to open serial device.\n") ;
-		return;
+		return NULL;
 	}
 	while(1){
 		if(serialDataAvail (fd_serial) ){ //읽을 데이터가 존재한다면,
@@ -186,7 +187,7 @@ void* bluetooth(void* argv){
 
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
 
-void open(){
+void open_door(){
 	pthread_mutex_lock(&lock_door);
 	oled(2);
 	lock(1);
@@ -194,7 +195,7 @@ void open(){
 	pthread_mutex_unlock(&lock_door);
 }
 
-void close(){
+void close_door(){
 	pthread_mutex_lock(&lock_door);
 	lock(1);
 	door(1);
@@ -277,7 +278,7 @@ void oled(int stat){
 }
 
 void init(){
-	done = false;
+	done = 0;
 	pthread_mutex_init(&lock_brightness, NULL);
 	pthread_mutex_init(&lock_dist, NULL);
 	pthread_mutex_init(&lock_done, NULL);

--- a/thread/main.c
+++ b/thread/main.c
@@ -314,9 +314,9 @@ int get_brightness(){
 }
 
 void set_distance(int value){
-	pthread_mutex_lock(&lock_brightness);
+	pthread_mutex_lock(&lock_dist);
 	distance = value;
-	pthread_mutex_unlock(&lock_brightness);
+	pthread_mutex_unlock(&lock_dist);
 }
 
 int get_distance(){

